src/main/java/org/opensearch/sdk/rest/BaseExtensionRestHandler.java

```diff
@@ -41,7 +44,9 @@
  */
 public abstract class BaseExtensionRestHandler implements ExtensionRestHandler {
 
-    private static String extensionName;
+    private static final String VALID_ROUTE_PREFIX_PATTERN = "^[a-zA-Z0-9-_]*$";
```

Comment:
1. Having a hyphen in the prefix is precisely the bug we are trying to fix here.
2. A hyphen inside square brackets in a regex must be the first or last character, or escaped.

Reply:
Hmm. when i test this regex pattern  ^[a-zA-Z0-9-_]*$ on regex101.com. It matches patterns as expected.

Matches: `hello-world123`, `hellWorld123`, `hello_world`
Doesn't match: `hello.world`, `hello/world`

Reply:
I will remove hyphen for it to be a subset of `^[a-zA-Z0-9:/*_]*$`

Reply:
> Hmm. when i test this regex pattern ^[a-zA-Z0-9-_]*$ on regex101.com. It matches patterns as expected.

Technically that is undefined behavior, so Regex101 is free do do whatever it wants to (which may or may not match what the JDK does).  From [the spec](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html):
> The interpretation of range expressions where the ending range point is also the starting range point of a subsequent range expression (for example, "[a-m-o]" ) is undefined.